ethergo/submitter/submitter.go

```
@@ -696,21 +765,35 @@
+		// if we failed to est gas for any reason, use *at least* the default flat gas on the next bump
+		gasLimit_fromEstimate = max(t.config.GetGasEstimate(chainID), gasLimit_fromEstimate)
 	}
 
-	return gasEstimate, nil
+	// start with whichever value is higher gas from the prior attempt, or gas from our latest simulation estimate
+	gasLimit_new = max(gasLimit_fromPrior, gasLimit_fromEstimate)
+
+	// whichever source is used as the base, multiply it by the configured gas unit add percentage
+	gasLimit_new = gasLimit_new + (gasLimit_new * uint64(gasUnitAddPercentage) / 100)
+	span.AddEvent("new gas limit", trace.WithAttributes(
+		attribute.Int64("gas_limit", int64(gasLimit_new)),
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion uint64 -> int64

(gosec)